packages/lib/src/components/UIElement.tsx

```diff
@@ -45,6 +48,25 @@ export class UIElement<P extends UIElementProps = any> extends BaseElement<P> im
         return state;
     }
 
+    /* eslint-disable-next-line */
+    protected submitAnalytics(obj = null) {
+        // Call analytics endpoint
+        let component = this.elementRef._id?.substring(0, this.elementRef._id.indexOf('-'));
+        if (component === 'dropin') {
+            const subCompID = this.elementRef['dropinRef'].state.activePaymentMethod._id;
+            component = `${component}-${subCompID.substring(0, subCompID.indexOf('-'))}`;
+        }
+
+        const aObj: AnalyticsObject = createAnalyticsObject({
+            class: 'log',
+            component,
+            type: ANALYTICS_SUBMIT_STR,
+            target: 'pay_button'
+        });
+
+        this.props.modules?.analytics.addAnalyticsAction('log', aObj);
+    }
```

Comment:
Good point

Reply:
Because the `onSubmit` function that gets called (which then calls `submitAnalytics`) is part of the _active payment method_ i.e. the `UIElement`, it is not part of `Dropin.tsx`